pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTValue.java

```diff
@@ -31,7 +31,7 @@
  * @author Andrew Bayer
  * @author Kohsuke Kawaguchi
  */
-public abstract class ModelASTValue extends ModelASTElement implements ModelASTMethodArg {
+public abstract class ModelASTValue extends ModelASTElement implements ModelASTMethodArg, ModelASTEnvironmentValue {
```

Comment:
Also, yes, `ModelASTInternalFunctionCall` does overlap to some extent with `ModelASTMethodCall`. The main difference is that `ModelASTInternalFunctionCall` doesn't allow recursive nesting of instances of itself as arguments, while `ModelASTMethodCall` does. Might want to rename/reorganize that at some point.